### PR TITLE
[4.0] [plg_system_httpheaders] Drop X-XSS-Protection and X-Content-Type-Options from the plugin

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #25735 && https://scotthelme.co.uk/security-headers-updates/

### Summary of Changes

Drop X-XSS-Protection and X-Content-Type-Options options from the plugin.

#### X-XSS-Protection

> As the most popular platform that still runs an XSS Auditor, Chrome announced plans to drop it in an Intent to Deprecate and Remove: XSS Auditor. There are further details available in this bug but the TLDR is that the XSS Auditor is going. Recent changes included taking it from block mode to filter mode by default and it seems that wasn't enough to save it from the false positives. Edge removed their XSS Filter last year so with the removal from Chrome there is now no browser that will have a native XSS protection. It's time for the header to go! With that said, and using my site as an example, you can see that the X-Xss-Protection header is no longer required for an A+ grade on Security Headers.

https://scotthelme.co.uk/security-headers-updates/

According to this the header is dying and was removed in Edge already and is now going to be remove in Chrome too. It makes no sense to ship Joomla 4 with that old thing that when we anyway only support the latest version of browsers. Based on that text it also seams that chrome in the past already defaulted to block mode now filter mode so older versions are also "protected" as long as you are using them. The best way to eliminate XSS is using CSP which we make much easier to do in Joomla 4 anyway.

#### X-Content-Type-Options

This header is now enforced by the default htaccess and web.config and seams to only confuse people as the switch has no affect at all anymore https://github.com/joomla/joomla-cms/issues/25735

### Testing Instructions

Apply this patch, make sure both headers are not possible to configure now and not set anymore.

### Expected result

both headers are not possible to configure now and not set anymore.

### Actual result

both headers are possible to configure now and get set as of today.

### Documentation Changes Required

Yes.

### Additional Info

I'm happy for all your input on this and this is a suggestion to deal with that two issues.

cc @SniperSister  :)

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 13f2cfd9fefbb07913488b7fb20043eeef2d7760<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

File: administrator/language/en-GB/en-GB.plg_system_httpheaders.ini [hunk 1]
```diff
@@ -37,9 +37,6 @@ PLG_SYSTEM_HTTPHEADERS_POSTINSTALL_INTRODUCTION_ACTION="Enable default security
 PLG_SYSTEM_HTTPHEADERS_REFERRERPOLICY="<a href='https://scotthelme.co.uk/a-new-security-header-referrer-policy/' target='_blank' rel='noopener noreferrer'>Referrer-Policy</a>"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.plg_system_httpheaders.ini:35]: Line 36, change `3` to `2`.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 4851e522376080d2bd43e45431f2aab33e129249<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by Quy** [comment]
Remove from the dropdown menu???

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 16b141e3104d8530b3566cc705f0fa2f3842586a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by sandewt** [comment]
screen shot 2019-08-03 at 17 29 47

(1) Supplement: 'X-Content-Type-Options' header is set as default in the htaccess or web.config file. (!?)

(2) How to set the 'X-Content-Type-Options' header if no use is made of htaccess or web.config? Or is this question not relevant?

File **httpheaders.php**, see line 68

```
private $supportedHttpHeaders = [
		'strict-transport-security',
		'content-security-policy',
		'content-security-policy-report-only',
		'x-frame-options',
		'referrer-policy',
		'expect-ct',
		'feature-policy',
	];
```
(3) The _feature-policy_ header may or may not supported !? Also view the image above.
	<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by zero-24** [comment]
> (1) Supplement: 'X-Content-Type-Options' header is set as default in the htaccess or web.config file. (!?)

I can not follow what you want me to-do?

> (2) How to set the 'X-Content-Type-Options' header if no use is made of htaccess or web.config? Or is this question not relevant?

Well here we have the same problem than for the there static headers. I can not detect whether the header is set or not using htaccess. And it seams the current workaround with offering that option still via the plugin is just to confusing. Any suggestions to implement such thing in a non-confusing way?

> (3) The feature-policy header may or may not supported !? Also view the image above.

It was just missing from the language string. Fixed.

**Comment by Quy** [comment]
> It was just missing from the language string. Fixed.

Lets remove that language string. It adds no additional information.

**Comment by sandewt** [comment]
> I can not follow what you want me to-do?

I mean, add this additional information to the language string.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by zero-24** [comment]
> Lets remove that language string. It adds no additional information.

Pushed.

> I mean, add this additional information to the language string.

Any suggestions about a text and where we should add it?

**Comment by sandewt** [comment]
> Any suggestions about a text and where we should add it?

Specify it in the documentation anyway.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by zero-24** [comment]
> Specify it in the documentation anyway.

👍

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on c9f8a85e664a2d1118e730065e2085ab222f809b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on c9f8a85e664a2d1118e730065e2085ab222f809b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25753">issues.joomla.org/tracker/joomla-cms/25753</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
Thanks!

**Comment by zero-24** [comment]
The two Headers have been removed from the docs: https://docs.joomla.org/J4.x:Http_Header_Management

Screenshot get updated as soon as other relevant PRs got merged.

**Pull request merged**